packages/sanes-chrome-extension/src/context/PersonaProvider.tsx

```diff
@@ -40,24 +41,26 @@ export const PersonaProvider = ({ children, persona }: Props): JSX.Element => {
   const [mnemonic, setMnemonic] = React.useState<string>(persona ? persona.mnemonic : '');
   const [txs, setTxs] = React.useState<ReadonlyArray<ProcessedTx>>(persona ? persona.txs : []);
   React.useEffect(() => {
-    const isExtensionContext = typeof chrome !== 'undefined';
-    if (!isExtensionContext) {
+    if (!extensionContext()) {
       return;
     }
 
     console.log('PersonaProvider registering listener');
-    chrome.runtime.onMessage.addListener((message, sender, _sendResponse) => {
-      if (sender.id !== chrome.runtime.id || !isMessageToForeground(message)) {
+    chrome.runtime.onMessage.addListener((msg, sender, _sendResponse) => {
+      const sameTarget = sender.id === chrome.runtime.id;
+      const msgToForeground = isMessageToForeground(msg);
+      const msgToPersonaProvider = msg.action === MessageToForegroundAction.TransactionsChanges;
+      if (!sameTarget || !msgToForeground || !msgToPersonaProvider) {
```

Comment:
Possible code duplication

Reply:
I think the question is: Should both providers listen to background messages individually? Or do we want one place that listens to messages from background and distributes them to the providers. Both ways are fine I think